cli/command/manifest/inspect.go

```diff
@@ -0,0 +1,119 @@
+package manifest
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/cli"
+	"github.com/docker/docker/cli/command"
+	"github.com/docker/docker/registry"
+)
+
+type inspectOptions struct {
+	remote   string
+	platform bool
+}
+
+// NewInspectCommand creates a new `docker manifest inspect` command
+func newInspectCommand(dockerCli *command.DockerCli) *cobra.Command {
+	var opts inspectOptions
+
+	cmd := &cobra.Command{
+		Use:   "inspect [OPTIONS] NAME[:TAG]",
+		Short: "Display an image's manifest, or a remote manifest list.",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.remote = args[0]
+			return runListInspect(dockerCli, opts)
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.BoolVarP(&opts.platform, "platform", "p", false, "Output additional info about platform")
+
+	return cmd
+}
+
+func runListInspect(dockerCli *command.DockerCli, opts inspectOptions) error {
+
+	// Get the data and then format it
+	var (
+		imgInspect []ImgManifestInspect
+		prettyJSON bytes.Buffer
+	)
+
+	named, err := reference.ParseNormalizedNamed(opts.remote)
+	if err != nil {
+		return err
+	}
+	targetRepo, err := registry.ParseRepositoryInfo(named)
+	if err != nil {
+		return err
+	}
+
+	// For now, always pull as there' no reason to store an inspect. They're quick to get.
+	// When the engine is multi-arch image aware, we can store these in a universal location to
+	// save a little bandwidth.
+	imgInspect, _, err = getImageData(dockerCli, named.Name(), "", true)
```

Comment:
👍 I thought for sure we had tests for that but I'm adding one, and adding this fix as well. Thanks @xiekeyang!